common/src/main/java/org/apache/druid/metadata/MetadataStorageConnectorConfig.java

```diff
@@ -78,6 +88,22 @@ public String getPassword()
     return passwordProvider == null ? null : passwordProvider.getPassword();
   }
 
+  public Properties getProperties() throws IOException
+  {
+    if (dbcpPropertiesFile == null) {
+      return null;
+    } else {
+      //no reason to save properties file as it is only used on startup
+      File propertiesFile = new File(dbcpPropertiesFile);
```

Comment:
Can we use `File.createTempFile` instead?

Reply:
Not creating any files here...